### PR TITLE
Enhanced init_tpm_parent(). It could use RSA algorithm

### PR DESCRIPTION
**Pull request by coolfengyu**
Enhanced init_tpm_parent(). It could use RSA algorithm to create primary key.

### CODE REVIEW

**Comment by diabonas** [comment]
Is RSA ever going to be used by this code? According to the [TCG PC Client Platform TPM Profile (PTP) Specification, Table 2](https://trustedcomputinggroup.org/wp-content/uploads/TCG_PC_Client_Platform_TPM_Profile_PTP_2.0_r1.03_v22.pdf#page=16), support for `TPM_ALG_ECC` is mandatory, so it will always be chosen for TPMs conforming to the specification.

**Comment by coolfengyu** [comment]
Some old TPM 2.0 devices don't support ECC, such as Infineon SLB9665TT2.0 TPM version 2.0.
This code will check the TPM device whether support ECC. If it supports, it use ECC. Otherwise, it will use RSA.

**Comment by AndreasFuchsSIT** [comment]
@coolfengyu which FW-version do you have ? Cause I'm pretty sure that all production-ready TPMs (also by IFX) should be supporting ECC.

**Comment by coolfengyu** [comment]
Intel NUC Board NUC5i5MYBE uses this TPM chip.
Some information of the chip:
TPM_PT_FAMILY_INDICATOR:                                                       
  as UINT32:                0x08322e3000                                       
  as string:                "2.0"                                              
TPM_PT_LEVEL:               0                                                  
TPM_PT_REVISION:            0.99                                               
TPM_PT_DAY_OF_YEAR:         0x000000ce                                         
TPM_PT_YEAR:                0x000007dd                                         
TPM_PT_MANUFACTURER:        0x49465800                                         
TPM_PT_VENDOR_STRING_1:                                                        
  as UINT32:                0x534c4239                                         
  as string:                "SLB9"                                             
TPM_PT_VENDOR_STRING_2:                                                        
  as UINT32:                0x36363500                                         
  as string:                "665"                                              
TPM_PT_VENDOR_STRING_3:                                                        
  as UINT32:                0x00000000                                         
  as string:                ""                                                 
TPM_PT_VENDOR_STRING_4:                                                        
  as UINT32:                0x00000000                                         
  as string:                ""                                                 
TPM_PT_VENDOR_TPM_TYPE:     0x00000001                                         
TPM_PT_FIRMWARE_VERSION_1:  0x00050000                                         
TPM_PT_FIRMWARE_VERSION_2:  0x00044102

File: src/tpm2-tss-engine-common.c [hunk 1]
```diff
@@ -462,9 +469,22 @@ init_tpm_parent(ESYS_AUXCONTEXT *eactx_p,
     r = Esys_TR_SetAuth(eactx_p->ectx, ESYS_TR_RH_OWNER, &ownerauth);
     ERRchktss(init_tpm_parent, r, goto error);
 
+    r = Esys_GetCapability (eactx_p->ectx,
+                            ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
+                            TPM2_CAP_ALGS, 0, TPM2_MAX_CAP_ALGS,
+                            NULL, &capabilityData);
+    ERRchktss(init_tpm_parent, r, goto error);
+
+    for (index = 0; index < capabilityData->data.algorithms.count; index++) {
```
Comment (AndreasFuchsSIT) [inline src/tpm2-tss-engine-common.c]: different signedness, index should be `UINT32` as count is.

File: src/tpm2-tss-engine-common.c [hunk 2]
```diff
@@ -462,9 +469,22 @@ init_tpm_parent(ESYS_AUXCONTEXT *eactx_p,
     r = Esys_TR_SetAuth(eactx_p->ectx, ESYS_TR_RH_OWNER, &ownerauth);
     ERRchktss(init_tpm_parent, r, goto error);
 
+    r = Esys_GetCapability (eactx_p->ectx,
+                            ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
+                            TPM2_CAP_ALGS, 0, TPM2_MAX_CAP_ALGS,
+                            NULL, &capabilityData);
```
Comment (AndreasFuchsSIT) [inline src/tpm2-tss-engine-common.c]: `capabilityData` must be free'd. We're leaking memory here.

**Pull request closed without merging**